### ISSUE TITLE
Warnings like: Cannot fetch state for io - Client error '403 Forbidden'

### ISSUE THREAD

**Issue by tne7laa** [maciej-or/hikvision_next#105]
So far my NVR (DS-7608NI-K2) is working in HA but seeing warnings like below. Don't if something is wrong. Can't figure it out.

Warning:
```
Logger: custom_components.hikvision_next.isapi
Source: custom_components/hikvision_next/isapi.py:804 
Integration: Hikvision NVR / IP Camera ([documentation](https://github.com/maciej-or/hikvision_next), [issues](https://github.com/maciej-or/hikvision_next/issues)) 
First occurred: 3 augustus 2023 om 11:45:17 (8620 occurrences) 
Last logged: 12:33:56

Unexpected exception | Cannot fetch state for io | Client error '403 Forbidden' for url 'http://192.168.1.8/ISAPI/System/IO/inputs/101' For more information check: https://httpstatuses.com/403
Unexpected exception | Cannot fetch state for io | Client error '403 Forbidden' for url 'http://192.168.1.8/ISAPI/System/IO/inputs/401' For more information check: https://httpstatuses.com/403
Unexpected exception | Cannot fetch state for motiondetection | Client error '403 Forbidden' for url 'http://192.168.1.8/ISAPI/ContentMgmt/InputProxy/channels/5/video/motionDetection' For more information check: https://httpstatuses.com/403
Unexpected exception | Cannot fetch state for videoloss | Client error '403 Forbidden' for url 'http://192.168.1.8/ISAPI/ContentMgmt/InputProxy/channels/5/video/videoLoss' For more information check: https://httpstatuses.com/403
```

Can I ignore these or do I need something to correct?

**Comment by maciej-or** [comment]
I see failed requests for alarm inputs 1 and 4 does it mean other (2, 3) work fine?
If no, check user permissions https://github.com/maciej-or/hikvision_next#hikvision-device-setup-checklist.
If yes it's really strange, please download and share diagnostic data form device info. It's safe all sensitive data are redacted.
![image](https://github.com/maciej-or/hikvision_next/assets/38075949/5ef4f6a9-9813-4df9-9eca-0c74b38c9498)
Basically you can ignore those warnings until you want to use alarm inputs.

**Comment by tne7laa** [comment]
Thanks for replying. It is an Hikvision NVR for 8 cameras. 7 are connected. 
Not all are Hikvision cameras connected. Don't know if they also should all be Hikvision. 
Camera 1 is an Chinese PTZ and camera 4 is an (older) Hikvision. 
At this moment I only use notifications (line crossing) from camera 3 (older Hikvision) and works Ok.
I don't know how it technically works. Enclosed the diagnostics from NVR.

I also can't find username and password I did put in somewhere for the HomeAssistant Integration. I have a special account for HA in NVR. Can I find this somewhere?

[config_entry-hikvision_next-3ad7f44696cabb7d0f9d361f8504c74d.json.txt](https://github.com/maciej-or/hikvision_next/files/12310759/config_entry-hikvision_next-3ad7f44696cabb7d0f9d361f8504c74d.json.txt)

**Comment by maciej-or** [comment]
It is an interesting case, I have never seen that before. Diagnostic data will help a lot in further analysis.
The integration speaks with NVR only so brand or age of cameras doesn't matter. You might check if NVR firmware is up to date.
Regarding  user, password and all data you put during configuration you will find them in `homeassitant/config/.storage/core.device_registry` file. Just search for 'hikvision_next' phrase.

**Comment by gumby2** [comment]
I believe this is the same error I'm getting for the cube camera;
see thread https://github.com/maciej-or/hikvision_next/issues/36

<ResponseStatus xmlns="http://www.hikvision.com/ver20/XMLSchema" version="2.0">
<requestURL>/ISAPI/ContentMgmt/InputProxy/channels</requestURL>
<statusCode>4</statusCode>
<statusString>Invalid Operation</statusString>
<subStatusCode>invalidOperation</subStatusCode>
</ResponseStatus>

**Comment by tne7laa** [comment]
> Regarding user, password and all data you put during configuration you will find them in `homeassitant/config/.storage/core.device_registry` file. Just search for 'hikvision_next' phrase.

Can't find any username/passoword in the core.device_registry file I used for configuring the Integration. I wanted to check (maybe change) if these are correct.

Only:

```
     {                                     
        "area_id": "werkkamer",             
        "config_entries": [                 
          "3ad7f44696cabb7d0ff8504c74d"
        ],                        
        "configuration_url": null,
        "connections": [       
          [                    
            "mac",                          
            "ac:cb:xx:fe:xx:90"
          ]                                      
        ],                                       
        "disabled_by": null,                     
        "entry_type": null,                      
        "hw_version": null,                          
        "id": "0dea1dbe6f3f0c058c362973ba",    
        "identifiers": [                             
          [                                          
            "hikvision_next",                        
            "DS-7608NI-K20820200829CC6067WCVU"
          ]                              
        ],                               
        "manufacturer": "Hikvision",     
        "model": "DS-7608NI-K2",         
        "name_by_user": null,              
        "name": "Network Video Recorder",
        "sw_version": "V4.32.115",
        "via_device_id": null               
}, etc
```

**Comment by maciej-or** [comment]
sorry @tne7laa I meant `homeassitant/config/.storage/core.config_entries file`

**Comment by gumby2** [comment]
On my cube camera /ISAPI/System/IO/inputs/1 works, but /ISAPI/System/IO/inputs/101 doesn't.  
/ISAPI/ContentMgmt/IOProxy/inputs doesn't work either, it seems there is no ContentMgmt at all for this camera, so no DigitalAlarmInputInfo, only AnalogAlarmInputInfo?  Like I have said, the 'Hikvision' integration does work for alarm input.

Let me now If I can assist in debugging the issue.  thanks for all your efforts!!

Scott

**Comment by maciej-or** [comment]
Hi @gumby2 I saw you shared diagnostic data in another thread. Thank you, it's enough for now. Before I start fixing bugs I need to write unit tests for the integration. It will allow to simulate every case without physical access to the particular device. By publishing this integration I did not suppose that so many devices have inconsistencies in their ISAPI responses.

**Comment by gumby2** [comment]
Glad to help!

**Comment by MaBeniu** [comment]
If there could be any value of additional info I would like to share my setup which also is resulting in exactly same error.
I have 8 cameras - 4 hik cameras and 4 cheap imou cams connected to 1 hik NVR (DS-7608NI-E2/8P).
I get error only for one hik camera (DS-2CD2442FWD-IW) which is registered under 6 position.
Error:  Cannot fetch state for io | Client error '403 Forbidden' for url 'http://IP.IP.IP.IP/ISAPI/System/IO/inputs/601'
